pkg/controller/backup_controller.go

```diff
@@ -865,3 +877,134 @@ func encodeToJSONGzip(data interface{}, desc string) (*bytes.Buffer, []error) {
 
 	return buf, nil
 }
+
+// deleteVolumeSnapshot delete VolumeSnapshot created during backup, if its
+// This is used to avoid deleting namespace in cluster triggers the VolumeSnapshot deletion,
+// which will cause snapshot deletion on cloud provider, then backup cannot restore the PV.
+// If DeletionPolicy is Retain, just delete it. If DeletionPolicy is Delete, need to
+// change DeletionPolicy to Retain before deleting VS, then change DeletionPolicy back to Delete.
+func (c *backupController) deleteVolumeSnapshot(volumeSnapshots []*snapshotv1api.VolumeSnapshot,
```

Comment:
Is it possible that we create a copy of VSC with `DeletionPolicy` as `Retain` and use this VSC to do the backup? In this way, we don't need to know the implementation details(e.g. patch the `Source` and `VolumeSnapshotRef`, the implementation of this part may change)

Reply:
Considered that way in implementation. After discussion with Daniel, I think keeping the DeletionPolicy as original may be more rational, because what we do here is to keep the Snapshot on cloud provider safe and have a trace for  deletion, and remove the namespaced resource VolumeSnapshot. There should be no more changed for resources already created.
Change DeletionPolicy from Delete to Retain will have some behavior change, although Retain is more safe, but it may be user's expected behavior.

Reply:
I mean create a copy of the VSC, there should be no change to the original one

Reply:
If we do that way, then need to update VS's referenced VSC to work, and may also consider the new VSC's status change after creation.
I think that can work, but need to add related logic. To me, the two ways both work, but no one is much better than the other.